ckanext/recombinant/write_excel.py

```diff
@@ -137,25 +140,44 @@ def excel_template(dataset_type, org):
         sheet.sheet_state = 'hidden'
     return book
 
+
 def append_data(book, record_data, chromo):
 
     """
     fills rows of an openpyxl.Workbook with selected data from a datastore resource
-
     """
     sheet = book[chromo['resource_name']]
     current_row = DATA_FIRST_ROW
     for record in record_data:
         for col_num, field in template_cols_fields(chromo):
-            item = record[field['datastore_id']]
-            if isinstance(item, list):
-                item = u', '.join(unicode(e) for e in item)
+            item = datastore_type_format(record[field['datastore_id']], field['datastore_id'], chromo)
             sheet.cell(row=current_row, column=col_num).value = item
         current_row += 1
 
     return book
 
 
+def datastore_type_format(value, datastore_id, chromo):
+
+    value_type = [f['datastore_type'] for f in chromo['fields'] if f['datastore_id'] == datastore_id]
```

Comment:
this function is called for every single cell, better to get the fields outside of the function and zip() them with the values instead of adding an O(N^2) operation like this

Reply:
actually, you already have the `field` in the loop above, pass that instead of chromo